### ISSUE TITLE
Segmentation Error when trying to create cluster

### ISSUE THREAD

**Issue by tomklino** [kubernetes/kops#7876]
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**
Version 1.14.0 (git-d5078612f)

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**
NA

**3. What cloud provider are you using?**
aws

**4. What commands did you run?  What is the simplest way to reproduce this issue?**
kops create -f test-cluster-spec.yaml
kops create secret --name test.nvsrc.com sshpublickey admin -i ~/.ssh/id_rsa.pub
kops update cluster test.nvsrc.com --yes

**5. What happened after the commands executed?**
```
tomk@LTLVTOMK ~/workspace/nuvo/kops/test (master) $ kops update cluster test.nvsrc.com --yes
W1104 12:15:33.456866    8959 urls.go:145] Using nodeup location from NODEUP_URL env var: "https://spotinst-public.s3.amazonaws.com/integrations/kubernetes/kops/v1.12.1-02a32434b/nodeup/linux/amd64/nodeup"
I1104 12:15:38.890078    8959 executor.go:103] Tasks: 0 done / 118 total; 52 can run
I1104 12:15:40.814846    8959 vfs_castore.go:729] Issuing new certificate: "apiserver-aggregator-ca"
I1104 12:15:40.858862    8959 vfs_castore.go:729] Issuing new certificate: "etcd-peers-ca-main"
I1104 12:15:40.954239    8959 vfs_castore.go:729] Issuing new certificate: "etcd-clients-ca"
I1104 12:15:40.968285    8959 vfs_castore.go:729] Issuing new certificate: "etcd-peers-ca-events"
I1104 12:15:41.017974    8959 vfs_castore.go:729] Issuing new certificate: "ca"
I1104 12:15:41.065412    8959 vfs_castore.go:729] Issuing new certificate: "etcd-manager-ca-main"
I1104 12:15:41.207932    8959 vfs_castore.go:729] Issuing new certificate: "etcd-manager-ca-events"
I1104 12:15:43.275272    8959 executor.go:103] Tasks: 52 done / 118 total; 33 can run
I1104 12:15:45.171290    8959 vfs_castore.go:729] Issuing new certificate: "apiserver-proxy-client"
I1104 12:15:45.281643    8959 vfs_castore.go:729] Issuing new certificate: "kops"
I1104 12:15:45.295031    8959 vfs_castore.go:729] Issuing new certificate: "kubelet"
I1104 12:15:45.367352    8959 vfs_castore.go:729] Issuing new certificate: "kube-proxy"
I1104 12:15:45.486027    8959 vfs_castore.go:729] Issuing new certificate: "aws-iam-authenticator"
I1104 12:15:45.490843    8959 vfs_castore.go:729] Issuing new certificate: "master"
I1104 12:15:45.624492    8959 vfs_castore.go:729] Issuing new certificate: "apiserver-aggregator"
I1104 12:15:45.635237    8959 vfs_castore.go:729] Issuing new certificate: "kube-scheduler"
I1104 12:15:45.732924    8959 vfs_castore.go:729] Issuing new certificate: "kubecfg"
I1104 12:15:45.735130    8959 vfs_castore.go:729] Issuing new certificate: "kube-controller-manager"
I1104 12:15:45.822612    8959 vfs_castore.go:729] Issuing new certificate: "kubelet-api"
I1104 12:15:48.184878    8959 executor.go:103] Tasks: 85 done / 118 total; 27 can run
I1104 12:16:03.018616    8959 executor.go:103] Tasks: 112 done / 118 total; 5 can run
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x29f1a07]

goroutine 1552 [running]:
k8s.io/kops/upup/pkg/fi.(*ResourceHolder).Unwrap(...)
	/go/src/k8s.io/kops/upup/pkg/fi/resources.go:232
k8s.io/kops/upup/pkg/fi.(*ResourceHolder).AsString(...)
	/go/src/k8s.io/kops/upup/pkg/fi/resources.go:237
k8s.io/kops/upup/pkg/fi/cloudup/spotinsttasks.(*Elastigroup).create(0xc00119c6c0, 0x41a5f40, 0xc0007cc2a0, 0x0, 0xc00119c6c0, 0xc0006f6d80, 0xc0007157b8, 0x48c5ac)
	/go/src/k8s.io/kops/upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go:446 +0xcf7
k8s.io/kops/upup/pkg/fi/cloudup/spotinsttasks.(*Elastigroup).createOrUpdate(0xc00119c6c0, 0x41a5f40, 0xc0007cc2a0, 0x0, 0xc00119c6c0, 0xc0006f6d80, 0x7fb18d634008, 0x0)
	/go/src/k8s.io/kops/upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go:338 +0xc5
k8s.io/kops/upup/pkg/fi/cloudup/spotinsttasks.(*Elastigroup).RenderAWS(0xc00119c6c0, 0xc00134e470, 0x0, 0xc00119c6c0, 0xc0006f6d80, 0x0, 0x0)
	/go/src/k8s.io/kops/upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go:333 +0x87
reflect.Value.call(0x39f4500, 0xc00119c6c0, 0x1a13, 0x3ab92f2, 0x4, 0xc00134c540, 0x4, 0x4, 0xc000aebe08, 0x13, ...)
	/usr/local/go/src/reflect/value.go:447 +0x461
reflect.Value.Call(0x39f4500, 0xc00119c6c0, 0x1a13, 0xc00134c540, 0x4, 0x4, 0xc00119c6c0, 0x1a13, 0xc001594840)
	/usr/local/go/src/reflect/value.go:308 +0xa4
k8s.io/kops/upup/pkg/fi.(*Context).Render(0xc000b379a0, 0x40d43a0, 0x0, 0x40d43a0, 0xc00119c6c0, 0x40d43a0, 0xc0006f6d80, 0x0, 0x0)
	/go/src/k8s.io/kops/upup/pkg/fi/context.go:223 +0xf16
k8s.io/kops/upup/pkg/fi.DefaultDeltaRunMethod(0x40d43a0, 0xc00119c6c0, 0xc000b379a0, 0xc0009b3201, 0xc000d6ec60)
	/go/src/k8s.io/kops/upup/pkg/fi/default_methods.go:79 +0x538
k8s.io/kops/upup/pkg/fi/cloudup/spotinsttasks.(*Elastigroup).Run(0xc00119c6c0, 0xc000b379a0, 0xc000d6ec60, 0x0)
	/go/src/k8s.io/kops/upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go:314 +0x41
k8s.io/kops/upup/pkg/fi.(*executor).forkJoin.func1(0xc00011cc80, 0x5, 0x5, 0xc000965580, 0xc000903ba0, 0xc000390c40, 0x3)
	/go/src/k8s.io/kops/upup/pkg/fi/executor.go:179 +0x1d1
created by k8s.io/kops/upup/pkg/fi.(*executor).forkJoin
	/go/src/k8s.io/kops/upup/pkg/fi/executor.go:175 +0xfb
```

**6. What did you expect to happen?**
the cluster to be created

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**
NA

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
[create-test-cluster-v10.txt](https://github.com/kubernetes/kops/files/3803814/create-test-cluster-v10.txt)

**9. Anything else do we need to know?**
I've added the following lines into the spec lately:
```
  kubeAPIServer:
    runtimeconfig:
      settings.k8s.io/v1alpha1:
    admissionControl:
      - PodPreset
```
May or may not be related.

**Comment by tomklino** [comment]
After digging around a bit more, I tracked the issue into a file "upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go" where the issue exists in v.1.14.0 but fixed in v.15.0-beta.1:

Line causing error in v1.14.0:
https://github.com/kubernetes/kops/blob/1.14.0/upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go#L446

Line fixing error in v1.15.0-beta.1:
https://github.com/kubernetes/kops/blob/1.15.0-beta.1/upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go#L535

So when I updated to v1.15.0-beta.1 I passed this issue, but ran into another, which that my cluster spec has `apiVersion: kops/v1alpha2` but it states in the changes of 1.15 that this was changes to `apiVersion: kops.k8s.io/v1alpha2`, that might be okay for a new cluster, but this is a test cluster, only meant to check before doing on an existing 1.14

Is there any chance that there will be a patch release for 1.14 that includes the fix on spotinsttasks ?

@liranp I see that you are a very relevant person for the file in question. Do you maybe have any insights on this?

**Comment by liranp** [comment]
@tomklino Sorry for the delay in replying. Please find below the download links for release 1.14.1 with cherry-picked bug fixes:

- [darwin_amd64](https://spotinst-public.s3.amazonaws.com/integrations/kubernetes/kops/v1.14.1-8aeefa9a4/darwin/amd64/kops)
- [linux_amd64](https://spotinst-public.s3.amazonaws.com/integrations/kubernetes/kops/v1.14.1-8aeefa9a4/linux/amd64/kops)
- [windows_amd64](https://spotinst-public.s3.amazonaws.com/integrations/kubernetes/kops/v1.14.1-8aeefa9a4/windows/amd64/kops.exe)

Please let us know if you have any further issues. Thanks!

**Comment by rifelpet** [comment]
seeing as the reported issue was fixed in 1.15 which has been stable for a while, i'm going to close this out.
/close
